data-products/src/main/scala/org/sunbird/analytics/audit/AssessmentScoreCorrectionJob.scala

```diff
@@ -0,0 +1,187 @@
+        AssessmentCorrectionMetrics(batchId = batchId, contentId = contentId, invalidRecords = 0, totalAffectedUsers = 0, contentTotalQuestions = contentMeta.totalQuestions)
+      }
+    }
+  }
+
+  def correctData(assessmentDF: DataFrame,
+                  userEnrolDF: DataFrame,
+                  batchId: String,
+                  contentId: String,
+                  totalQuestions: Int,
+                  isDryRunMode: Boolean
+                 )(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig, sc: SparkContext): AssessmentCorrectionMetrics = {
+    val incorrectRecords: DataFrame = assessmentDF.filter(col("content_id") === contentId).filter(col("total_max_score") =!= totalQuestions)
+      .select("course_id", "batch_id", "content_id", "attempt_id", "user_id", "total_max_score", "total_score")
+    val incorrectRecordsSize: Long = incorrectRecords.count()
+    JobLogger.log("Total Incorrect Records", Option(Map("total_records" -> incorrectRecordsSize)), INFO)
+    if (incorrectRecordsSize > 0) {
+      removeAssessmentRecords(incorrectRecords, batchId, isDryRunMode)
+      generateInstructionEvents(incorrectRecords, batchId, contentId)
+      saveRevokingCertIds(incorrectRecords, userEnrolDF, batchId)
+    }
+    AssessmentCorrectionMetrics(batchId = batchId, contentId = contentId, invalidRecords = incorrectRecordsSize, totalAffectedUsers = incorrectRecords.select("user_id").distinct().count(), contentTotalQuestions = totalQuestions)
+  }
+
+
+  def saveRevokingCertIds(incorrectRecords: DataFrame,
+                          userEnrolmentDF: DataFrame,
+                          batchId: String)(implicit config: JobConfig): Unit = {
+    val certIdsDF: DataFrame = incorrectRecords.select("course_id", "batch_id", "content_id", "user_id").distinct()
+      .join(userEnrolmentDF, incorrectRecords.col("user_id") === userEnrolmentDF.col("userid") && incorrectRecords.col("course_id") === userEnrolmentDF.col("courseid") &&
+        incorrectRecords.col("batch_id") === userEnrolmentDF.col("batchid"), "left_outer")
+      .withColumn("certificate_data", explode_outer(col("issued_certificates")))
+      .withColumn("certificate_id", col("certificate_data.identifier"))
+      .select("courseid", "batchid", "userid", "certificate_id").filter(col("certificate_id") =!= "")
+    saveLocal(certIdsDF, batchId, "revoked-cert-data")
+  }
+
+  def saveLocal(data: DataFrame,
+                batchId: String,
+                folderName: String)(implicit config: JobConfig): Unit = {
+    JobLogger.log("Generating the CSV File", None, INFO)
+    val outputPath = config.modelParams.getOrElse(Map[String, Option[AnyRef]]()).getOrElse("csvPath", "").asInstanceOf[String]
+    data.repartition(1).write.option("header", value = true).format("com.databricks.spark.csv").save(outputPath.concat(s"/$folderName-$batchId-${System.currentTimeMillis()}.csv"))
```

Comment:
Please use the contentId also in the name of the file. If a course has multiple SelfAssess content, it will be difficult to understand for which content the file generated.